src/components/single-product-page/SingleProductCard.js

```diff
@@ -1,12 +1,54 @@
 import './SingleProductCard.css';
+import { Link } from 'react-router-dom';
+
+import { useWishlist, useAuth, useAuthModal } from '../../hooks';
 
 const SingleProductCard = props => {
-  const { title, brand, imageUrl, price, discount } = props.product;
+  const { showModal } = useAuthModal();
+  const { isAuth } = useAuth();
+  const { wishlist, removeFromWishlist, addToWishlist } = useWishlist();
+  const {
+    title,
+    brand,
+    imageUrl,
+    price,
+    discount,
+    _id: prodId,
+  } = props.product;
+
+  const toggleWishListHandler = prodId => {
+    if (!isAuth) return showModal();
+
+    addToWishlist(prodId);
+
+    wishlist.items.includes(prodId)
+      ? removeFromWishlist(prodId)
+      : addToWishlist(prodId);
+  };
+
+  const wishlistButton =
+    isAuth && wishlist.items.includes(prodId) ? (
+      <button
+        onClick={toggleWishListHandler.bind(null, prodId)}
```

Comment:
I'm not sure why bind is being used here. any explanations?

Reply:
I am using it here for binding the argument. Here there no "this" keyword available so I am passing null as the first argument.